test/components/beta/auto_complete_test.rb

```diff
@@ -84,85 +80,39 @@ def test_forwards_input_name_on_parent_to_input_slot_name
   end
 
   def test_renders_with_icon
-    render_inline Primer::Beta::AutoComplete.new(label_text: "Fruits", src: "/url", list_id: "my-list-id", input_id: "test-input", with_icon: true)
-
-    assert_selector('auto-complete[for="my-list-id"][src="/url"]') do
-      assert_selector("input.form-control")
-      assert_selector('label[for="test-input"]')
-      assert_selector("svg.octicon.octicon-search")
-    end
-  end
-
-  def test_renders_screen_reader_label_with_icon_and_non_visible_label
-    render_inline Primer::Beta::AutoComplete.new(label_text: "Fruits", src: "/url", list_id: "my-list-id", input_id: "test-input", is_label_visible: false, with_icon: true)
-
-    assert_selector('auto-complete[for="my-list-id"][src="/url"]') do
-      assert_selector("input.form-control")
-      assert_selector('label[for="test-input"]')
-      assert_selector("svg.octicon.octicon-search")
-      assert_selector("span.sr-only", text: "Fruits")
-    end
-  end
-
-  def test_wraps_icon_and_input_when_with_icon_true
-    render_inline Primer::Beta::AutoComplete.new(label_text: "Cool Fruits", src: "/url", input_id: "test-input", list_id: "my-list-id", with_icon: true)
+    render_inline Primer::Beta::AutoComplete.new(label_text: "Fruits", src: "/url", list_id: "my-list-id", input_id: "test-input") do |component|
+      component.leading_visual_icon(icon: :search)
```

Comment:
I think missing end?
```suggestion
      component.leading_visual_icon(icon: :search)
    end
```